### PR TITLE
Pr dpdk compilefix

### PR DESCRIPTION
**Pull request by asedlmayr**
Fix for compiling with the USE_DPDK flag.

Changed the #ifdef to #if defined() because defined can be used in compound expressions (but not necessary in these cases).
Updated the dpdk functions to align with our coding standards.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[![Build Status](https://falcon.ecg.rice.edu:443/buildStatus/icon?job=github_public_agora%2Fdevelop)](https://falcon.ecg.rice.edu:443/job/github_public_agora/job/develop/)
+[![Build Status](https://falcon.ecg.rice.edu:443/buildStatus/icon?job=github_public_agora%2Fpr-dpdk-compilefix)](https://falcon.ecg.rice.edu:443/job/github_public_agora/job/pr-dpdk-compilefix/)
```
Comment (asedlmayr) [inline README.md:1]:
```suggestion
[![Build Status](https://falcon.ecg.rice.edu:443/buildStatus/icon?job=github_public_agora%2Fdevelop)](https://falcon.ecg.rice.edu:443/job/github_public_agora/job/develop/)
```

File: README.md [hunk 2]
```diff
@@ -94,8 +94,8 @@ We provide a high performance [packet generator](simulator) to emulate the RRU.
    * In another terminal, run  `./build/sender --num_threads=2 --core_offset=1 --frame_duration=5000 --enable_slow_start=1 --conf_file=data/tddconfig-sim-ul.json` to start the emulated RRU 
      with uplink configuration.
    * The above steps use Linux networking stack for packet I/O. Agora also supports using DPDK
-   to bypass the kernel for packet I/O. To enable DPDK, run `cmake -DUSE_DPDK=1 ..; make -j` to 
-   rebuild code for Mellanox NICs; for Intel NICs, run `cmake -DUSE_DPDK=1 -DUSE_MLX_NIC=0 ..; make -j` 
+   to bypass the kernel for packet I/O. To enable DPDK, run `cmake .. -DUSE_DPDK=1; make -j` to 
+   rebuild code for Mellanox NICs; for Intel NICs, run `cmake .. -DUSE_DPDK=1 -DUSE_MLX_NIC=0; make -j`
```
Comment (asedlmayr) [inline README.md:98]: Not necessary but works either way.  I changed this to force the CI to run again, so moving back to the original is not a bad idea either.

Comment (jianding17) [reply README.md:98]: Changing it back to the original might be better to be consistent with the rest of README.

File: README.md [hunk 3]
```diff
@@ -94,8 +94,8 @@ We provide a high performance [packet generator](simulator) to emulate the RRU.
    * In another terminal, run  `./build/sender --num_threads=2 --core_offset=1 --frame_duration=5000 --enable_slow_start=1 --conf_file=data/tddconfig-sim-ul.json` to start the emulated RRU 
      with uplink configuration.
    * The above steps use Linux networking stack for packet I/O. Agora also supports using DPDK
-   to bypass the kernel for packet I/O. To enable DPDK, run `cmake -DUSE_DPDK=1 ..; make -j` to 
-   rebuild code for Mellanox NICs; for Intel NICs, run `cmake -DUSE_DPDK=1 -DUSE_MLX_NIC=0 ..; make -j` 
+   to bypass the kernel for packet I/O. To enable DPDK, run `cmake .. -DUSE_DPDK=1; make -j` to
```
Comment (asedlmayr) [inline README.md:97]:
```suggestion
   to bypass the kernel for packet I/O. To enable DPDK, run `cmake -DUSE_DPDK=1 ..; make -j` to 
```

Comment (asedlmayr) [inline README.md:97]:
```suggestion
   to bypass the kernel for packet I/O. To enable DPDK, run `cmake -DUSE_DPDK=1 ..; make -j` to 
```

File: README.md [hunk 4]
```diff
@@ -95,7 +95,7 @@ We provide a high performance [packet generator](simulator) to emulate the RRU.
      with uplink configuration.
    * The above steps use Linux networking stack for packet I/O. Agora also supports using DPDK
    to bypass the kernel for packet I/O. To enable DPDK, run `cmake -DUSE_DPDK=1 ..; make -j` to 
-   rebuild code for Mellanox NICs; for Intel NICs, run `cmake -DUSE_DPDK=1 -DUSE_MLX_NIC=0 ..; make -j` 
+   rebuild code for Mellanox NICs; for Intel NICs, run `cmake .. -DUSE_DPDK=1 -DUSE_MLX_NIC=0; make -j`
```
Comment (asedlmayr) [inline README.md:98]:
```suggestion
   rebuild code for Mellanox NICs; for Intel NICs, run `cmake -DUSE_DPDK=1 -DUSE_MLX_NIC=0 ..; make -j` 
```

**Pull request merged**